Support building ubi images in CI

mdeggies opened the pull request.

mdeggies commented on .github/workflows/build.yml:
```diff
@@ -20,33 +20,34 @@ jobs:
         run: |
           echo "Building with Go $(cat .go-version)"
-          echo "::set-output name=go-version::$(cat .go-version)"
+          echo "go-version=$(cat .go-version)" >> $GITHUB_OUTPUT
```
This is just some cleanup that gets rid of those set-output errors in the actions UI

mdeggies commented on .github/workflows/build.yml:
```diff
@@ -20,33 +20,34 @@ jobs:
           VERSION="${{ github.event.inputs.version || '0.0.0-dev' }}"
           echo "Using version ${VERSION}"
-          echo "::set-output name=product-version::${VERSION}"
+          echo "product-version=${VERSION}" >> $GITHUB_OUTPUT
```
This is just some cleanup that gets rid of a couple set-output errors in the actions UI

mdeggies commented on .github/workflows/build.yml:
```diff
@@ -117,3 +118,24 @@ jobs:
+          version: ${{env.version}}
+          target: ubi
+          arch: amd64
+          redhat_tag: scan.connect.redhat.com/ospid-60f9fdbec3a80eac643abedf/${{env.repo}}:${{env.version}}-ubi
```
As commented, this tag will need to change to `quay.io/redhat-isv-containers/5f89bbc30b94cf64cfeb500b/${{env.repo}}:${{env.version}}-ubi` on October 20th

samsalisbury replied:
Red Hat have made their changes live, the tag format will now be as follows (they've dropped the product name from the tags entirely, and use the project ID not the old `ospid-...`):

```suggestion
          redhat_tag: quay.io/redhat-isv-containers/5f89bbc30b94cf64cfeb500b:${{env.version}}-ubi
```

cc @mdeggies

mdeggies replied:
Thanks, Sam! I committed your suggestion.

tvoran commented on Dockerfile:
```diff
@@ -48,6 +59,59 @@ COPY dist/$TARGETOS/$TARGETARCH/vault-k8s /bin/
+ARG TARGETOS TARGETARCH
+
+# Set ARGs as ENV so that they can be used in ENTRYPOINT/CMD
+ENV PRODUCT_VERSION=$PRODUCT_VERSION
+ENV BIN_NAME=$BIN_NAME
```
It looks like these were already defined on lines 72-73?

mdeggies replied:
Oh good catch, ty!

tvoran reviewed (commented):
This passed the openshift preflight for me, and our vault-helm injector acceptance tests. I think the ospid isn't quite right (but of course that tag should be changing tomorrow anyway).

I think we'll probably want to remove the `prod-ubi-image` target in the Makefile as well.

mdeggies commented:
Thanks for the review! I removed the make target and removed those duplicates lines in the dockerfile. Lmk if you notice anything else -- I confirmed the project ID in redhat, and the necessary secrets (registry key + project ID) have been set in vault.
